### DISCUSSION TITLE
Any plans on implementing SWD/SMD (by extension MUS) files found in certain NDS games?

### DISCUSSION THREAD

**Discussion by Russsst** [vgmtrans/vgmtrans#440] [Music formats]
I'd understand if not, as its not even close to being as common as SDAT in NDS games.
Even more so for the MUS format that's just an archive for the SWD and SMD files (and as far as I know is used in Soma Bringer and one other game).

I've tried manually ripping the files from Soma bringer myself, but the only files usable were some streamed audio.
Dumping rom into audacity as raw data game some samples, but most noise.

Completely understandable though if it doesn't happen considering this seems to be mostly a volunteer driven project

**Comment by sykhro** [comment]
Are there any existing documents about this driver?

**Reply by Russsst** [reply]
Since the DS Pokemon Mystery Dungeon games use the DSE sound driver, I was able to find a bit of documentation with a quick search
This seems to give detailed info on the driver and the formats: 
https://projectpokemon.org/docs/mystery-dungeon-nds/procyon-studios-digital-sound-elements-r12/

(I found this through their repo where they used this research https://github.com/PsyCommando/ppmdu_2?tab=readme-ov-file)

Here an archive of the website from procyon, though its all in japanese
https://web.archive.org/web/20161207013202/https://www.procyon-studio.co.jp/dse/

as for the MUS format

Finding anything to do with it is hard,

The best I've gotten (in terms of somabringer's case is this fourm post
https://www.hcs64.com/mboard/forum.php?showthread=22127&showpage=5

This then lead to a dead repo where they at least documented _some_ stuff on the format
https://github.com/loveemu/psf-drivers/blob/master/2sf/Soma%20Bringer/NTR-YBSJ-JPN.txt

It is still the DSE driver, just seems to be a package format for the most part

Took a while to find all this again since its been a year. I'm not expecting a ton since its volunteer work, but I'll happily provide what I've found since I'm not a programmer.

**Reply by sykhro** [reply]
sorry, I was tired and misread your post. I have recently pushed a draft implementation of DSE on this branch https://github.com/vgmtrans/vgmtrans/commits/feature/dse-pmd/
if you feed it SWD and then SMD it should extract stuff. there's a bunch of things left to polish/fix, but that's better than nothing :)
